vaadin-date-time-picker-flow-parent/vaadin-date-time-picker-flow/src/main/java/com/vaadin/flow/component/datetimepicker/DateTimePicker.java

```diff
@@ -292,11 +292,21 @@ private void synchronizeChildComponentValues(LocalDateTime value) {
     }
 
     /**
-     * Adds the given Component to the specified slot of this component.
+     * Removes the web components default slot contents, and adds the given
+     * component to it.
      */
-    private void addToSlot(Component component, String slot) {
+    private void replaceDefaultSlotContent(String slot, Component component) {
         Objects.requireNonNull(component, "Component to add cannot be null");
         component.getElement().setAttribute("slot", slot);
+        component.getElement().setAttribute("is-flow-slotted-child", "");
+
+        // executeJs call to remove the default slot contents. executeJs
+        // actually runs after appendChild, so the removal logic would also
+        // remove the replacement component we add below. To fix that we check
+        // for a custom attribute and only remove children that do not have it.
+        getElement().executeJs(
+                "this.querySelectorAll(`[slot=\"${$0}\"]`).forEach(child => !child.hasAttribute('is-flow-slotted-child') && child.remove());",
+                slot);
         getElement().appendChild(component.getElement());
```

Comment:
This is pretty dumb, but I could not find a way to synchronize the executeJs and appendChild calls.

I was thinking about adding a mode to the web component to prevent it from creating its own default content by setting an attribute, but I think that doesn't work with templates, as the creation of the element is controlled by the user's template.

Reply:
Actually dedicated methods on the web component for replacing the date picker and time picker might work.